File: golang/lc00001-lc00100/lc00003/longest_substring_without_repeating_characters_test.go
package lc00003

import "testing"

type caseType struct {
	str      string
	expected int
}

func TestLengthOfLongestSubstring(t *testing.T) {
	cases := []caseType{
		{"abcabcbb", 3},
		{"bbbbb", 1},
		{"pwwkew", 3},
		{"", 0},
	}

	for _, c := range cases {
		result := lengthOfLongestSubstring(c.str)
		if result != c.expected {
			t.Fatalf("lengthOfLongestSubstring function failed, str: %s, expected: %d, result: %d", c.str, c.expected, result)
		}
	}
}

File: php/lc00001-lc00100/lc00003/LongestSubstringWithoutRepeatingCharactersTest.php
<?php

namespace leetcode\lc00003;

use PHPUnit\Framework\TestCase;

class LongestSubstringWithoutRepeatingCharactersTest extends TestCase
{
    public function testLengthOfLongestSubstring()
    {
        $cases = [
            ["str" => "abcabcbb", "expected" => 3],
            ["str" => "bbbbb", "expected" => 1],
            ["str" => "pwwkew", "expected" => 3],
            ["str" => "", "expected" => 0],
        ];

        $obj = new LongestSubstringWithoutRepeatingCharacters();
        for ($i = 0; $i < count($cases); $i++) {
            $this->assertEquals($cases[$i]['expected'], $obj->lengthOfLongestSubstring($cases[$i]['str']));
        }
    }
}
